examples/scripts/reward_trainer_summarize.py

```diff
@@ -0,0 +1,229 @@
+    pad_to_multiple_of: Optional[int] = None
+    return_tensors: str = "pt"
+
+    def __call__(self, features):
+        # features_chosen = []
+        # features_rejected = []
+        merged_features = []
+        for feature in features:
+            # check if the keys are named as expected
+            if (
+                "input_ids_chosen" not in feature
+                or "input_ids_rejected" not in feature
+                or "attention_mask_chosen" not in feature
+                or "attention_mask_rejected" not in feature
+            ):
+                raise ValueError(
+                    "The features should include `input_ids_chosen`, `attention_mask_chosen`, `input_ids_rejected` and `attention_mask_rejected`"
+                )
+
+            merged_features.append(
+                {
+                    "input_ids": feature["input_ids_chosen"],
+                    "attention_mask": feature["attention_mask_chosen"],
+                }
+            )
+            merged_features.append(
+                {
+                    "input_ids": feature["input_ids_rejected"],
+                    "attention_mask": feature["attention_mask_rejected"],
+                }
+            )
+        batch = self.tokenizer.pad(
+            merged_features,
+            padding=self.padding,
+            max_length=self.max_length,
+            pad_to_multiple_of=self.pad_to_multiple_of,
+            return_tensors=self.return_tensors,
+        )
+        batch = {
+            "input_ids": batch["input_ids"],
+            "attention_mask": batch["attention_mask"],
+            "return_loss": True,
+        }
+        return batch
+
+
+
+def create_and_prepare_model(args):
+
+    torch_dtype = torch.float32
+
+    model = AutoModelForSequenceClassification.from_pretrained(
+        args.model_name,
+        num_labels=1,
+        torch_dtype=torch_dtype,
+    )
+
+    model.config.torch_dtype = torch_dtype
+    model.config.use_cache = True
+
+
+    tokenizer = AutoTokenizer.from_pretrained(args.model_name)
+    if getattr(tokenizer, "pad_token", None) is None:
+        tokenizer.pad_token = tokenizer.eos_token
+
+    if getattr(model.config, "pad_token_id", None) is None:
+        model.config.pad_token_id = model.config.eos_token_id
+
+    return model, tokenizer
+
+
+def create_and_prepare_dataset(args, tokenizer, split, num_proc=2):
+    dataset = load_dataset(args.dataset_name, split=split)
+    original_columns = dataset.column_names
+
+    def preprocess_function(examples):
+        new_examples = {
+            "input_ids_chosen": [],
+            "attention_mask_chosen": [],
+            "input_ids_rejected": [],
+            "attention_mask_rejected": [],
+        }
+        for prompt, chosen, rejected in zip(examples["prompt"], examples["chosen"], examples["rejected"]):
+            tokenized_chosen = tokenizer(
+                prompt + "\n\n" + chosen, padding="max_length", truncation=True, max_length=args.seq_length
```

Comment:
so it turns out the chosen and rejected have two spaces after TLDR: instead of one.  It's not a big issue but you can remove it with `chosen = chosen[:6] + chosen[7:]` and same with rejected.

Reply:
probably just keep as is for now